### ISSUE TITLE
Docker with Systemd - ExecStop not working during container Stop

### ISSUE THREAD

**Issue by subbu165** [moby/moby#36787]
I have been working on running systemd within a container and have customer service start/stop defined as part of systemd service.
I had tried hard to fix this up and I couldn't do so. 

So I had systemd running inside docker like below (DockerFile content):
```Dockerfile
#Setting up systemd
ENV container docker
RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == systemd-tmpfiles-setup.service ] || rm -f $i; done); \
rm -f /lib/systemd/system/multi-user.target.wants/*;\
rm -f /etc/systemd/system/*.wants/*;\
rm -f /lib/systemd/system/local-fs.target.wants/*; \
rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
rm -f /lib/systemd/system/basic.target.wants/*;\
rm -f /lib/systemd/system/anaconda.target.wants/*;
VOLUME [ "/sys/fs/cgroup" ]
CMD ["/usr/sbin/init"]
```

I have a service file saved like below in /etc/systemd/system/test.service (inside the container)
```
[Unit]
Description=Setup Platform

[Service]
Type=oneshot
RemainAfterExit=true
ExecStart=/bin/echo "My Test Start!"
ExecStop=/bin/echo "My Test Stop!"

[Install]
WantedBy=multi-user.target
```

So now every time the container starts, the "My Test Start!" is called successfully. But where as every time when container stops, I could never see the 'My Test Stop!', which means ExecStop never been called during container stop. I have googled Extensively on this issue for the past 2 days. Tried various work arounds, nothing seems to be working. I came across one post too (https://developers.redhat.com/blog/2016/09/13/running-systemd-in-a-non-privileged-container/), where they have said, docker stop sends SIGTERM which is not good enough for systemd to shutdown and we should actually configure stop signal to '--stop-signal SIGRTMIN+3', even this doesn't work for me.

I'm using `FROM centos:7.4.1708`, systemd 219 and Docker version 17.12.1-ce, build 7390fc6. I have been slogging it out for past 2/3 days without any success.

I'm using `journalctl -u test.service -f` to see the logs and to watch out for "my Test stop' to be printed before the container exits. But container exits without printing the 'My Test Stop'. This is how I'm verifying.

**Comment by thaJeztah** [comment]
This looks related to a check in runc that prevented `SIGRTMIN` signals being sent to the container's process. This check was removed in https://github.com/opencontainers/runc/pull/1706

Comparing the version of runc that's used in Docker 17.12.1 with runc 1.0.0-rc5, it looks like that change is not yet in the version used by Docker 17.12.1; https://github.com/opencontainers/runc/compare/9f9c96235cc97674e935002fc3d78361b696a69e...v1.0.0-rc5

Docker 18.03.1 does ship with runc v1.0.0-rc5, so does have this fix; can you try upgrading docker to 18.03.1, and see if that resolves your issue?

**Comment by thaJeztah** [comment]
Looks like this is a duplicate of https://github.com/moby/moby/issues/36219, which was addressed through https://github.com/moby/moby/pull/36222

Let me go ahead and close this issue, but let me know if you still run into this on a current version of Docker
